qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,335 @@
+
+    Where x, y or z are binary variables, and P is penalty constant. In this class,
+    value of P is automatically determined, but can be supplied as argument at the timing
+    of instantiation.
+
+    If the constraint does not match the pattern of classical constraint, the constraint
+    is kept as is. Otherwise, constraint is converted into equivalent penalty and added
+    to objective function.
+
+    References:
+        [1]: Fred Glover, et al. (2019),
+             A Tutorial on Formulating and Using QUBO Models,
+             `arXiv:1811.11538 <https://arxiv.org/abs/1811.11538>`_.
+
+    """
+
+    def __init__(self, penalty: Optional[float] = None) -> None:
+        """
+        Args:
+            penalty: Penalty factor to scale equality constraints that are added to objective.
+                     If None is passed, a penalty factor will be automatically calculated on
+                     every conversion.
+        """
+
+        self._src_num_vars = None  # type: Optional[int]
+        self._dst = None  # type: Optional[QuadraticProgram]
+        self.penalty = penalty  # type: Optional[float]
+
+    def convert(self, problem: QuadraticProgram) -> QuadraticProgram:
+        """Convert a problem of known inequality constraints into an unconstrained problem.
+
+        Args:
+            problem: The problem to be solved, that does not contain inequality constraints.
+
+        Returns:
+            The converted problem
+
+        Raises:
+            QiskitOptimizationError: If an unsupported-type variable exists.
+        """
+
+        # create empty QuadraticProgram model
+        self._src_num_vars = problem.get_num_vars()
+        self._dst = QuadraticProgram(name=problem.name)
+
+        # If no penalty was given, set the penalty coefficient by _auto_define_penalty()
+        if self._should_define_penalty:
+            penalty = self._auto_define_penalty(problem)
+        else:
+            penalty = self._penalty
+
+        # Set variables
+        for x in problem.variables:
+            if x.vartype == Variable.Type.CONTINUOUS:
+                self._dst.continuous_var(x.lowerbound, x.upperbound, x.name)
+            elif x.vartype == Variable.Type.BINARY:
+                self._dst.binary_var(x.name)
+            elif x.vartype == Variable.Type.INTEGER:
+                self._dst.integer_var(x.lowerbound, x.upperbound, x.name)
+            else:
+                raise QiskitOptimizationError("Unsupported vartype: {}".format(x.vartype))
+
+        # get original objective terms
+        offset = problem.objective.constant
+        linear = problem.objective.linear.to_dict()
+        quadratic = problem.objective.quadratic.to_dict()
+        sense = problem.objective.sense.value
+
+        # convert linear constraints into penalty terms
+        for constraint in problem.linear_constraints:
+
+            # special constraint check function here
+            if not self._is_special_constraint(problem, constraint):
```

Comment:
It might be better to mention "target constraint" instead of "special constraint" to clarify that such constraints are to be converted.

Reply:
Changed to `_is_matched_constraint`